src/placement/jump_map.c

```diff
@@ -365,30 +367,34 @@ get_target(struct pool_domain *curr_dom, struct pool_target **target,
 			end_tgt = start_tgt + (num_doms - 1);
 
 			range_set = isset_range(tgts_used, start_tgt, end_tgt);
-			if (range_set)
-				clrbit_range(tgts_used, start_tgt, end_tgt);
+			if (range_set) {
+				/* Used up all targets in this domain */
+				setbit(dom_occupied, curr_dom - root_pos);
+				D_ASSERT(top != -1);
+				curr_dom = dom_stack[top--]; /* try parent */
+				continue;
+			}
```

Comment:
Is this code needed here? This "return to parent" thing should never happen, because of the code block below.

Reply:
Resolved via Skype, just thinking about it.